Userland/Libraries/LibSoftGPU/Device.h

```diff
@@ -66,6 +68,14 @@ struct RasterizerOptions {
     bool cull_front { false };
     u8 texcoord_generation_enabled_coordinates { TexCoordGenerationCoordinate::None };
     Array<TexCoordGenerationConfig, 4> texcoord_generation_config {};
+    bool lighting_enabled { false };
+};
+
+struct LightModelParameters {
+    FloatVector4 scene_ambient_color { 0.2f, 0.2f, 0.2f, 1.0f };
+    bool viewer_at_infinity { false };
+    unsigned int single_color;
```

Comment:
Where are `viewer_at_infinity` and `single_color` used?

Reply:
They're not currently as I haven't _quite_ implemented all of the possible light model stuff. The `viewer_at_infinity` var is used for specular highlights. `single_color` controls whether or not the specular part is calculated as a "second_color" (I have no idea what this means in practice)

Reply:
Ah, maybe `(void)`-use them somewhere with a `FIXME` comment then? Then it's clear what those two fields are doing

Reply:
Please at least default initialize it?